src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,158 @@
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.time.format.DateTimeFormatter;
+
+// TODO: Add a test for the constructor.
+
+/** A generator of a collection of AvailabilityTimeSlot Objects. */
+public class AvailabilityTimeSlotGenerator {
+  private static final int numberOfSlotsPerDay = 48;
+  private static final int TIME_8AM = 8;
+  private static final int TIME_7PM = 19;
+
+  /**
+   * Constructs a list of a day's worth of AvailabilityTimeSlot objects.
+   *
+   * @param instant An instant on the day that time slots are generate for.
+   * @param timezoneOffset A String that represents the difference between UTC and the user's
+   *     current timezone. Example: A user in EST has a timezoneOffset of "-240" which means that
+   *     EST is 240 minutes behind UTC.
+   */
+  // TODO: Create a timeSlotsForWeek method.
+  public static List<AvailabilityTimeSlot> timeSlotsForDay(Instant instant, String timezoneOffset) {
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZoneId zoneId = day.getZone();
+    String dayOfWeek = day.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.US);
+    int year = day.getYear();
+    int month = day.getMonthValue();
+    int dayOfMonth = day.getDayOfMonth();
+
+    String date = generateDate(dayOfWeek, month, dayOfMonth);
+    List<String> times = generateTimes();
+    List<String> utcEncodings = generateUTCEncodings(year, month, dayOfMonth, zoneId);
+    List<Boolean> selectedStatuses = getSelectedStatuses();
+    return generateTimeSlots(utcEncodings, times, date, selectedStatuses);
+  }
+
+  // Uses an Instant and a timezoneOffset String to create a ZonedDateTime instance
+  // for the day specified by the Instant.
+  private static ZonedDateTime generateDay(Instant instant, String timezoneOffset) {
+    ZoneOffset timeZoneOffset = convertStringToOffset(timezoneOffset);
+    ZoneId zoneId = ZoneId.ofOffset("UTC", timeZoneOffset);
+    return instant.atZone(zoneId);
+  }
+
+  // This method takes the timezoneOffset String and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertStringToOffset(String timezoneOffset) {
+    int offsetTotalMinutes = Integer.parseInt(timezoneOffset);
+    int offsetHours = offsetTotalMinutes / 60;
+    int offsetMinutes = offsetTotalMinutes % 60;
+    return ZoneOffset.ofHoursMinutes(offsetHours, offsetMinutes);
+  }
+
+  // Returns a readable date string such as "Tue 7/7".
+  private static String generateDate(String dayOfWeek, int month, int dayOfMonth) {
+    return dayOfWeek + " " + month + "/" + dayOfMonth;
+  }
+
+  // This method returns a list of UTC Strings for a single day, creating one String per Time
+  // Slot.
+  private static List<String> generateUTCEncodings(
+      int year, int month, int dayOfMonth, ZoneId zoneId) {
+    List<String> utcEncodings = new ArrayList<String>();
+    int[] minutes = {0, 15, 30, 45};
+    for (int i = TIME_8AM; i <= TIME_7PM; i++) {
```

Comment:
`    for (int i = TIME_8AM; i <= TIME_7PM; i++) {` is little better than `    for (int i = 8; i <= 19; i++) {`

The intent of the "use a constant" comment is for you to capture the semantics of the arguments. My suggestions in line with that are
`    for (int i = EARLIEST_HOUR; i <= LATEST_HOUR; i++) {`

One reason this is better is for the simple reason that you can change the value of the constant without making the constant's name become a lie.

Reply:
Done.